### PR TITLE
Break SigninView into smaller component

### PR DESCRIPTION
**Pull request by gemscng**
This should be a better approach. First step before customize signin view.

### CODE REVIEW

File: admin/client/Signin/components/UserInfo.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { Button} from 'elemental';
+
+
+var UserInfo = React.createClass({
+	render: function(){
+		if (!this.props.user) return null;
+		const openKeystoneButton = this.props.userCanAccessKeystone ? <Button href={Keystone.adminPath} type="primary">Open Keystone</Button> : null;
+		return (
+			<div className="auth-box__col">
+			<p>Hi {this.props.user.name.first},</p>
```
Comment (mxstbr) [inline admin/client/Signin/components/UserInfo.js]: Nit: Proper indentation here

File: admin/client/Signin/components/LoginForm.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Button, Form, FormField, FormInput } from 'elemental';
+
+var LoginForm = React.createClass({
+	render: function() {
+		if (this.props.user) return null;
+		return (
+			<div className="auth-box__col">
+			<Form method="post" onSubmit={this.props.handleSubmit} noValidate>
```
Comment (mxstbr) [inline admin/client/Signin/components/LoginForm.js]: Nit: proper indentation here

File: admin/client/Signin/components/AlertView.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	render: function(){
+		if (this.props.isInvalid) {
+			return <Alert key="error" type="danger" style={{ textAlign: 'center' }}>{this.props.invalidMessage}</Alert>;
+		} else if (this.props.signedOut) {
+			return <Alert key="signed-out" type="info" style={{ textAlign: 'center' }}>You have been signed out.</Alert>;
+		} else {
+			/* eslint-disable react/self-closing-comp */
+			// TODO: This probably isn't the best way to do this, we
+			// shouldn't be using Elemental classNames instead of components
+			return <div key="fake" className="Alert Alert--placeholder">&nbsp;</div>;
```
Comment (mxstbr) [inline admin/client/Signin/components/AlertView.js]: Instead of this return `null` here, this used to not be possible but should be fine now.

**Comment by mxstbr** [comment]
This looks awesome, thanks so much @gemscng! Just two small nitpicks with JSX indentation and we're good to go! 

How did you find the new architecture to work with?

**Comment by gemscng** [comment]
@mxstbr The new architecture is pretty good. The component structure using reactjs make customization a breeze. 

Right now, i am using following method to do customization. Any better suggestion? 

`````````
	var bundles = {
		singin: browserify(global.__base + './../../../../adminuiroutes/client/signin/index.js')
	};
	
	// prebuild static resources on the next tick
	// improves first-request performance
	process.nextTick(function () {
		bundles.singin.build();
	});
	app.all('/' + keystone.get('admin path') + '/js/signin.js', bundles.singin.serve);
``````````


Customized Signin View.

``````````
class FreeplaySigninView extends SigninView{
  render() {

    const boxClassname = classnames('auth-box', {
      'auth-box--has-errors': this.state.isAnimating,
    });
    return (
      <div className="auth-wrapper">
        <AlertView 
          isInvalid = {this.state.isInvalid}
          signedOut = {this.state.signedOut}
          invalidMessage={this.state.invalidMessage}
        />
        <div className = {boxClassname}>
          <h1 className="u-hidden-visually">{this.props.brand ? this.props.brand : 'Keystone'} Sign In </h1>
          <div className="auth-box__inner">
            <Brand 
              logo = {this.props.isInvalid}
              brand = {this.props.brand}
            />
            <UserInfo 
              user = {this.props.user}
              userCanAccessKeystone = {this.props.userCanAccessKeystone}
            />
            <LoginForm
              user = {this.props.user}
              handleSubmit = {this.handleSubmit}
              handleInputChange = {this.handleInputChange}
              email = {this.state.email}
              password = {this.state.password}
              animating = {this.state.animating}
            />
            <div> Hello World</div>  //Added a 
          </div>
        </div>
        <div className="auth-footer">
          <span>Powered by </span>
          <a href="http://keystonejs.com" target="_blank" title="The Node.js CMS and web application platform (new window)">KeystoneJS</a>
        </div>
      </div>
    );
  }
};


ReactDOM.render(
	<FreeplaySigninView />,
	document.getElementById('signin-view')
);
``````````

**Comment by mxstbr** [comment]
This looks perfect, thanks so much for taking the time! ✨ Great work!

**Comment by mxstbr** [comment]
To answer your question, that looks fine, but all of these changes are being made to make customizing the admin interface a lot easier. Prepare to see improvements on that front soon!

**Comment by gemscng** [comment]
Ic. Anyway we have a project need to customize adminui and also reuse the component in other front end ui. I will take this approach first. 😄

**Comment by mxstbr** [comment]
Awesome, glad you managed to work it out—thanks so much for helping out here!

**Pull request merged**